client/state/jetpack-connection-health/selectors/is-jetpack-connection-problem.js

```diff
@@ -19,3 +23,53 @@ export default function isJetpackConnectionProblem( state, siteId ) {
 
 	return connection_data.jetpack_connection_problem;
 }
+
+/**
+ * Hook to check if the current site has possible Jetpack connection problem.
+ *
+ * Returns true if the current site has possible Jetpack connection problem
+ *
+ * @param  {number}  siteId        Site ID
+ */
+export const useIsJetpackConnectionProblem = ( siteId ) => {
+	const dispatch = useDispatch();
+	const isPossibleConnectionProblem = useSelector( ( state ) => {
+		return isJetpackConnectionProblem( state, siteId );
+	} );
+
+	useEffect( () => {
+		const onMessage = ( event ) => {
+			const error = event.data?.[ 0 ];
+			const status = event.data?.[ 1 ];
+			if ( error?.message && status > 200 ) {
+				if ( error.message.includes( 'site is inaccessible' ) && ! isPossibleConnectionProblem ) {
+					dispatch( setJetpackConnectionMaybeUnhealthy( siteId ) );
+				}
+			}
+		};
+
+		/**
+		 * This event listener is used to detect if the site is inaccessible.
+		 * Among others it's used as a login in `wpcom-proxy-request` package,
+		 * and the idea is borrowed from there to detect the site inaccessibility.
+		 */
+		window.addEventListener( 'message', onMessage );
+
+		return () => {
+			window.removeEventListener( 'message', onMessage );
+		};
+	}, [ isPossibleConnectionProblem, siteId, dispatch ] );
```

Comment:
I was trying to find other alternatives to this global listener, but I cannot imagine another way to do it.

Reply:
Yes, I searched all Jetpack reducer stuff before I ended up with that solution. I ended up enabling debug and noticed that the only consistent thing was the `wpcom-proxy-request` package. 
Searching a bit realized that the messages I saw were introduced using the same way I followed here.